### DISCUSSION TITLE
How to handle null in nested property bindings?

### DISCUSSION THREAD

**Discussion by moon6969** [AvaloniaUI/Avalonia#7857] [Q&A]
When SelectedThing is null,
    `<TextBox Text="{Binding Path=SelectedThing.NestedClass.TestString, Converter={StaticResource NullStringConverter}, Mode=OneWay}" />`
gives a Binding error:
_'Null value in expression '{empty}' at 'SelectedThing'.'_
and the converter is not called.

The only workaround I've managed is to define TestString directly on the ViewModel since this allows the converter to be called... but it's getting messy.

Is there a better way?

**Comment by maxkatz6** [comment]
There are two more properties in Binding: TargetNullValue and FallbackValue. You can use them to handle nulls or errors.

**Reply by moon6969** [reply]
Thanks Max,

I do use a FallbackValue, but it doesn't prevent the Binding error appearing in the log.
What is the impact if there are a huge number of binding errors like this?

I was really hoping there was some sweet Avalonia sugar like a XAML null-conditional operator to pass control to the converter to provide some context relevant fallback values in code.
EG:
`Path=SelectedThing?.NestedClass?.TestString`

**Reply by lostmsu** [reply]
@maxkatz6 just facing this, I think this deserves to be a feature request.

**Reply by Wiratudomsamut** [reply]
-

**Comment by deerchao** [comment]
And for a null collection, it would be nice to be able to specify an empty collection:

`<ListBox ItemSource="{Binding A.B.C,FallbackValue=[]}" />`

or even better

`<ListBox ItemSource="{Binding A?.B?.C,FallbackValue=[]}" />`

**Comment by Jannik-M91** [comment]
I just had the same issue. While not a real solution, I worked around it be defining a static "dummy" variant of whatever is assigned to "SelectedThing".

When nothing is selected I would assign the dummy object instead which has default values for all members (e.g. "NestedClass").

I found this to be cleaner than having to implement everything directly in the View Model.

Anyway: +1 for the null-conditional operator in bindings!

**Reply by maxkatz6** [reply]
Null-conditional operators is a good idea. But somebody need to create this feature request issue :)

**Reply by Jannik-M91** [reply]
Done, see https://github.com/AvaloniaUI/Avalonia/issues/17029
